### DISCUSSION TITLE
What recommended UI components for Next.js?

### DISCUSSION THREAD

**Discussion by yb** [vercel/next.js#39844] [Help] [answered]
Hi all, 
I am trying to use Next.js to make a website. 
I want to know which UI components are compatible with Next.js, and I have tried some frameworks before this:

- **Ant Design**: This must be considered by every React developer. But if you use it with Next.js, you need to configuration the less loader and others. Even with `next-plugin-antd-less`, the compiled CSS file will be ~200KB (without gzip), it contains all styles for components that are not used.
- **Evergreen UI**:  Although its official website is developed using Next.js ([source code](https://github.com/segmentio/evergreen/tree/master/docs)), but I still encountered a lot of compatibility issues when using it, which seems to be related to server-side rendering.

Can you recommend some UI frameworks or components that are compatible with Next.js? It is better if the bundle size is optimized.

Thanks

**Comment by icyJoseph** [comment, marked as answer]
Hi,

No short, nor straight forward answer here. 

Pretty much most frameworks work just fine, unless they do not account for SSR, in general not just Next.js, that is to say that they do unguarded DOM access or window checks, which would break when run on the server. 

 If you are just after some name, I'd say try out [Mantine](https://mantine.dev/guides/next/), but again, I'd reinforce the idea that any framework ought to work more or less correctly, unless, designed to only work with CSR.

Most frameworks require work to be done in the `_document` page, where all stylesheets are collected and packaged as part of the initial HTML sent to the client. 

I've also tried `vanilla-extract` and tailwind with various degrees of satisfaction, however, these are CSS solutions rather than fleshed out component libraries. Also styled-components of course. 

I'd personally recommend to take a headless/agnostic UI library, for example Reach-UI, Radix, or headless-ui itself, and custom-style that yourself. A step further would be to use https://react-spectrum.adobe.com/react-aria/, which gives a11y behaviour, and lets your style up your own components. I am a big CSS fan, so I might be biased here.

**Reply by yb** [reply]
Thanks @icyJoseph, this is a very helpful answer. I've tried Mantine - it should be the UI framework I'm looking for.

**Reply by icyJoseph** [reply]
Cool! I'd ask to mark ad answered, but I'm sure a framework recommendation is really an "answer". Have fun!

**Comment by Yashraj950** [comment]
https://jamsr-ui.jamsrworld.com/components/accordion


try this

**Reply by yb** [reply]
In 2024, I have fully used [shadcn/ui](https://ui.shadcn.com/) - it is almost perfectly compatible with next.js.
Thanks for your recommendation.
